logstash-core/lib/logstash/pipeline.rb

```diff
@@ -100,7 +100,9 @@ def initialize(config_str, settings = SETTINGS, namespaced_metric = nil)
       raise
     end
     @queue = build_queue_from_settings
-    @input_queue_client = @queue.write_client
+
+    @input_queue_client_factory = LogStash::Instrument::WriteClientFactory.new(queue.write_client, self, metric)
```

Comment:
TBH I don't think we need a factory at all. If we had multiple implementations of the factory *or* if the factory had complicated construction logic I'd say it'd be a good idea, but as is, i'd just do the wrapping here.